inc/zoo/AnyCallable.h

```diff
@@ -65,8 +66,60 @@ struct AnyCallable<TypeErasureProvider, R(Args...)>: TypeErasureProvider {
                 const_cast<AnyCallable &>(*this)
             );
     }
+
+    void swap(AnyCallable& other) noexcept {
```

Comment:
This is correct, but I prefer to not trust the type erasure provider to give `noexecept` swap, I will change this to `noexcept(noexcept(declval<TypeEraser>().swap(declval<TypeEraser>()))`